config/file_config.go

```diff
@@ -43,6 +48,40 @@ type samplerConfigType struct {
 	Sampler string
 }
 
+type InMemoryCollectorConfig struct {
+	// CacheCapacity must be less than math.MaxInt32
+	CacheCapacity int `validate:"required,lt=2147483647"`
+}
+
+type HoneycombLevel int
+
+type HoneycombLoggerConfig struct {
+	LoggerHoneycombAPI string         `validate:"required,url"`
+	LoggerAPIKey       string         `validate:"required"`
+	LoggerDataset      string         `validate:"required"`
+	Level              HoneycombLevel `validate:"required"`
+}
+
+type PrometheusMetricsConfig struct {
+	MetricsListenAddr string `validate:"required"`
+}
+
+type HoneycombMetricsConfig struct {
+	MetricsHoneycombAPI      string `validate:"required,url"`
+	MetricsAPIKey            string `validate:"required"`
+	MetricsDataset           string `validate:"required"`
+	MetricsReportingInterval int64  `validate:"required"`
+}
+
+type PeerManagementConfig struct {
+	Type                    string   `validate:"required,oneof= file redis"`
+	Peers                   []string `validate:"required,dive,url"`
```

Comment:
what does `dive` here mean?

Reply:
> This tells the validator to dive into a slice, array or map and validate that level of the slice, array or map with the validation tags that follow.

It validates that each string in the array is a URL.